system/integrations/checkmk/setup-guide.rst

```diff
@@ -0,0 +1,98 @@
+based on the message contents.
+
+That means that “setting up Checkmk integration”
+is simply a matter of **adding configuration to your Checkmk site**:
+telling it when to send these messages and what to say.
+To do that, you’ll need to create a new **notification rule**
+(see `Checkmk’s official docs <https://checkmk.com/cms_notifications.html>`_
+for help with that).
+
+As part of this new rule, you’ll have to choose a **notification method**
+(*i.e.,* a script to execute whenever the rule is triggered).
+This script will be written by you (samples below),
+and contain the logic for sending API/email messages to Zammad:
+
+.. figure:: /images/system/integrations/checkmk/adding-new-notification-rules.png
+   :alt: Checkmk "New Rule" dialog
+   :align: center
+
+   Find your custom scripts in the Checkmk WATO under **Notifications > New Rule > Notification Method**.
+
+Once you’re done setting up your new rule, you’re all set!
+New tickets should start coming in and auto-updating
+any time your rule is triggered.
+
+(If you need help troubleshooting, be sure to check the :ref:`checkmk-recent-logs`.)
+
+.. _checkmk-api-alerts:
+
+Scripts for API Alerts
+^^^^^^^^^^^^^^^^^^^^^^
+
+To add these scripts in the Checkmk WATO,
+copy them into your Checkmk installation directory and make them executable.
+(Be sure to replace the ``zammad.example.com`` callback URL
+with the one found in your admin panel.)
+
+Service notification
+   For updates on the status of the software running on your server
+   (*e.g.,* postgres).
+
+   .. code:: bash
+
+      #!/bin/bash
+      # /opt/omd/sites/<SITE>/local/share/check_mk/notifications/zammad-service
+
+      curl -X POST \
+        -F "event_id=$NOTIFY_SERVICEPROBLEMID" \
+        -F "host=$NOTIFY_HOSTNAME" \
+        -F "service=$NOTIFY_SERVICEDESC" \
+        -F "state=$NOTIFY_SERVICESTATE" \
+        -F "text=$NOTIFY_SERVICEOUTPUT" \
+        https://zammad.example.com/api/v1/... # see Admin Panel > System > Integrations > Checkmk > Usage
+
+Host notification
+   For updates on the status of the server itself.
+
+   .. code:: bash
+
+      #!/bin/bash
+      # /opt/omd/sites/<SITE>/local/share/check_mk/notifications/zammad-host
+
+      curl -X POST \
+        -F "event_id=$NOTIFY_HOSTPROBLEMID" \
+        -F "host=$NOTIFY_HOSTNAME" \
+        -F "state=$NOTIFY_HOSTSTATE" \
+        -F "text=$NOTIFY_HOSTOUTPUT" \
+        https://zammad.example.com/api/v1/... # see Admin Panel > System > Integrations > Checkmk > Usage
+
+.. note:: 🤔 **What’s with all the env vars?**
+
+   Whenever Checkmk runs these scripts,
+   it needs to provide some information
+   about the event that triggered the notification.
+   This information is passed in the form of
+   these ``$NOTIFY_*`` environment variables.
+
+   You can specify additional parameters to pass to the script
+   when you’re setting up your notification rule,
+   but the ones you see here are all provided by default.
+
+Scripts for Email Alerts
```

Comment:
This heading really bothers me.
Technically while checkmk does use a script under the hood, this will be a generic email based notification sent from checkmk. This means that there's no script involved.

I'd prefer a heading like e.g. `E-Mail notification based alerts` or something like that.
Because otherwise the question would be "how do I have to write my script for those mails??" which basically would be incorrect.

Reply:
TIL! Obviously I am new to Checkmk. Would you be cool with it if we changed both headings to the following?

* API Alerts
* Email Alerts

Reply:
This is addressed in the latest commit. Leaving this conversation open so that you can a) resolve it if you're okay with it, or b) continue the discussion if you're not.